introduction/07_questionnaires.rst

```diff
@@ -114,39 +162,130 @@ Examples
+    Students are then granted points also for partially correct answers. You can try it out 
+    below.
+    
+    For instance, in this case there are three correct answers, the grading goes as follows:
+    1 correct = 3 points,
+    2 correct = 6 points,
+    3 correct = 10 points.
+    
+    And for the wrong answers:
+    1 wrong option chosen = 3 points deducted, 
+    2 wrong options chosen = 6 points deducted,
+    3 wrong options chosen = 10 points deducted.
+
+    When :math:`(x + 1)^2 = 16`, what is :math:`x`?
+
+    a. 4
+    *b. an integer
+    *c. 3
+    d. an irrational number
+    e. -3
+    *f. -5
+
+Freetext questions
+------------------
 
 The ``freetext`` subdirective creates text boxes and grades based on their
 input.
 
-.. questionnaire:: questionnaire_text_demo 15
-  :title: A simple multiple-choice questionnaire
+The freetext directive also accepts the following options in addition to 
+the common question options:
+
+- ``length``: (horizontal) length for the HTML text input
+- ``height``: If greater than 1, the textarea HTML element is used. Otherwise, 
+  a text input is used.
+
+**Instructions and correct answer:** The body of the freetext question is 
+expected to be its model solution. However, the question instructions can be 
+written inside the body before the model answer. The instructions and the 
+model solution must be separated with an empty line.
+
+Note that if the questionnaire has the ``feedback`` option set, freetext questions 
+may not have a model solution and the body of the question is shown as the 
+question instructions.
+
+**Comparison method**: The freetext directive accepts a second positional 
+argument after the points. It defines the compare method for the model solution. 
+For instance ``freetext:: 5 int``. 
+
+The different comparison methods are ``int``, ``float``, ``string``, 
+``subdiff``, ``regexp`` or ``unsortedchars`` (unsorted character set). The ``regexp`` 
+compare method takes the correct answer as a regular expression and tries to
+match the submission with it. ``subdiff``: works almost like the string method, 
+but it can have multiple correct answers separated with ``|`` and if the answer is 
+incorrect, it shows the difference of the answer to each correct answer as a hint. 
+For example, when the correct answer is *'cat'* and the student answers *'car'*, 
+the student receives feedback: *"Correct parts in your answer: ca-."* 
+
+String methods have comparison modifiers that are separated with a hyphen. 
+For example, ``.. freetext:: 30 string-ignorews-ignorequotes``. 
+
+The following modifiers are available:
+
+- ``ignorews:`` ignore white space (applies to regexp too)
+- ``ignorequotes``: iqnore "quotes" around
+- ``requirecase``: require identical lower and upper cases (only with the 
+  string and subdiff types)
+- ``ignorerepl``: ignore REPL prefixes
+- ``ignoreparenthesis``: ignore parenthesis "( )"
+|
+
```

Comment:
Example title?. Again, showing both, the example and the code may be beneficial for the reader